numba/tests/test_array_methods.py

```diff
@@ -1329,6 +1335,45 @@ def test_imag(self):
         z = x + 1j*y
         np.testing.assert_equal(pyfunc(z), cfunc(z))
 
+    def test_clip(self):
+        # TODO: scalars are not tested (issue #3469)
+        for a in (np.linspace(-10, 10, 101),
+                  np.linspace(-10, 10, 40).reshape(5, 2, 4)):
+            for pyfunc in [np_clip, array_clip]:
+                cfunc = jit(nopython=True)(pyfunc)
+
+                with self.assertRaises(ValueError):
+                    pyfunc(a, None, None)
+                with self.assertRaises(ValueError):
+                    cfunc(a, None, None)
+
+                np.testing.assert_equal(pyfunc(a, 0, None), cfunc(a, 0, None))
+                np.testing.assert_equal(pyfunc(a, None, 0), cfunc(a, None, 0))
+
+                np.testing.assert_equal(pyfunc(a, -5, 5), cfunc(a, -5, 5))
+
+                pyout = np.empty_like(a)
+                cout = np.empty_like(a)
+                np.testing.assert_equal(pyfunc(a, -5, 5, pyout),
+                                        cfunc(a, -5, 5, cout))
+                np.testing.assert_equal(pyout, cout)
+
+                # This test exhibits undefined behaviour since min > max, but
+                # we'll make sure it's consistent with NumPy, which computes
+                # min(max(val, a_min), a_max) since 1.17. Prior to that min and
+                # max were the other way round, so we're not consistent with
+                # older NumPy versions.
```

Comment:
Should probably note this in the docs?

Reply:
Or maybe we shouldn't test this at all? I copied this test from the original PR (then had to modify it to match the current NumPy behaviour) - In this scenario, the user is doing something daft by trying to clip with a minimum greater than the maximum. I think perhaps it would be better to remove this check and in general not attempt to match NumPy when the user does strange things.

Are you happy if I just remove this check instead?

Reply:
As Numba 0.54 is moving to NumPy 1.17 as a minimum this probably becomes moot? I agree however that removing this test on the basis it's checking something strange is probably fine.

Reply:
I think it does. I've removed the test.

Reply:
Great, thanks.